### ISSUE TITLE
go get yield: invalid version: unknown revision 000000000000

### ISSUE THREAD

**Issue by TheoBrigitte** [prometheus-operator/prometheus-operator#3530]
**What happened?**

```
$ go get -u github.com/prometheus-operator/prometheus-operator
go: github.com/prometheus-operator/prometheus-operator upgrade => v0.42.1
go get: github.com/prometheus-operator/prometheus-operator@v0.42.1 requires
        github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

**Did you expect to see something different?**

I expect to be able to use `go get -u github.com/prometheus-operator/prometheus-operator` without any error.

**How to reproduce it (as minimally and precisely as possible)**:

1. Setup a project which imports `github.com/prometheus-operator/prometheus-operator/pkg/client/versioned`

###### go.mod
```
module test

go 1.15
```

###### main.go
```
package main

import (
        promclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
)

type Client struct {
        prom promclient.Interface
}
```

2. run `go get -u github.com/prometheus-operator/prometheus-operator`

**Environment**

```
$ go version
go version go1.15.2 linux/amd64
```

**Comment by paulfantom** [comment]
This should be solved with https://github.com/prometheus-operator/prometheus-operator/pull/3531. @TheoBrigitte could you confirm?

**Comment by TheoBrigitte** [comment]
This is broken in v0.42.1, so we need a new release with correct tags in place to be able to test this.

So far I can only test via commit sha, which produces a correct result.

```
$ go get -u github.com/prometheus-operator/prometheus-operator@fbd01683839aa408b31fa15fa1aa91c68f13d7ef
go: downloading github.com/prometheus-operator/prometheus-operator v0.42.2-0.20200928114327-fbd01683839a
go: github.com/prometheus-operator/prometheus-operator fbd01683839aa408b31fa15fa1aa91c68f13d7ef => v0.42.2-0.20200928114327-fbd01683839a
```

But I would consider this fixed when I can run the following command with no error

```
$ go get -u github.com/prometheus-operator/prometheus-operator
```

Currently this produces the exact same error.

**Comment by paulfantom** [comment]
Thank you for verification :+1:

**Comment by zetaab** [comment]
I am also hitting this

```
GO111MODULE=on go mod tidy
go: finding module for package github.com/prometheus-operator/prometheus-operator/pkg/client/versioned
go: found github.com/prometheus-operator/prometheus-operator/pkg/client/versioned in github.com/prometheus-operator/prometheus-operator v0.42.1
go: github.com/prometheus-operator/prometheus-operator@v0.42.1 requires
	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

**Comment by DenOlijf** [comment]
hi, any updates on this :)

**Comment by simonpasquier** [comment]
we need v0.43.0 to be released which should be soon...

**Comment by DenOlijf** [comment]
because of this:

v0.42.2-0.20200928114327-fbd01683839a/pkg/client/versioned/typed/monitoring/v1/alertmanager.go:90:5: too many arguments in call to c.client.Get().Namespace(c.ns).Resource("alertmanagers").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Do
	have (context.Context)
	want ()

?

**Comment by simonpasquier** [comment]
Closed by the release of [v0.43.0](https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.43.0). There's still an issue due to Cortex (AFAICT) so you'd need to use a `replace` directive to pin the correct version of `k8s.io/client-go` just like this repository does:

https://github.com/prometheus-operator/prometheus-operator/blob/9a39a3a90b6bf193c4b98e2cb86f6fec0ef532d0/go.mod#L43-L45
